### PR TITLE
Better decimal support in Scala generators

### PR DESCRIPTION
**Pull request by wing328**
- Better decimal support in scala generators

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

cc @clasnake (2017/07), @jimschubert (2017/09) ❤️, @shijinkui (2018/01), @ramzimaalej (2018/03), @chameleon82 (2020/03), @Bouillie (2020/04)

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaClientCodegen.java [hunk 1]
```diff
@@ -116,6 +116,7 @@ public ScalaAkkaClientCodegen() {
         importMapping.remove("List");
         importMapping.remove("Set");
         importMapping.remove("Map");
+        importMapping.put("BigDecimal", "java.math.BigDecimal");
```
Comment (chameleon82) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaClientCodegen.java:119]: Scala has it's own `scala.math.BigDecimal` type and global alias `BigDecimal`
https://www.scala-lang.org/api/2.12.0/scala/math/BigDecimal.html
Should we use java type? As those types can be interfered in the client code. Then I will have to write something like this:
```scala
import java.math.{ BigDecimal => JBigDecimal }
...
 val income:  BigDecimal  = db.getIncome(..)
 val outcome: JBigDecimal = generatedClient.getOutcome(..)
 val balance: BigDecimal  = income - BigDecimal(outcome)
```

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaClientCodegen.java:119]: We can use `scala.math.BigDecimal` if that's preferable.

I used `java.math.BigDecimal` in this change as it's copied from another Scala generator.

Let me know what you guys (Scala developers) prefer.

Comment (chameleon82) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaClientCodegen.java:119]: `scala.math.*` `BigInt` and `BigDecimal` are preferable types in Scala. No import required for that types.

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaClientCodegen.java]: Done via https://github.com/OpenAPITools/openapi-generator/pull/7876/commits/bd3a773b3fa968a7a6878409d317323ef49e6dbb. Please have another look when you've time. Thanks.

**Review by chameleon82 — APPROVED** [review]

**Pull request merged**